scripts/update.sh

```diff
@@ -3,60 +3,136 @@
 # Licensed under the MIT license. See LICENSE file on the project webpage for details.
 set -x
 
-EDX_ROLE=${1:-edxapp}
+EDX_ROLE="edxapp"
+DEPLOYMENT_ENV="dev"
+ACCESS_TOKEN=""
 
 OXA_TOOLS_PATH=/oxa/oxa-tools
 OXA_CONFIG_PATH=/oxa/configuration
 
+display_usage() {
+  echo "Usage: $0 -a|--access_token {access token} [-r|--role {mongo|mysql|edxapp|fullstack}] [-e|--environment {dev|bvt|int|prod}]"
+  exit 1
+}
+
+is_valid_arg() {
+  local list="$1"
+  local arg="$2"
+
+  if [[ $list =~ (^|[[:space:]])"$arg"($|[[:space:]]) ]] ; then
+    result=0
+  else
+    result=1
+  fi
+
+  return $result
+}
+
+parse_args() {
+  while [[ "$#" -gt 0 ]]
+  do
+    case "$1" in
+      -r|--role)
+        EDX_ROLE="${2,,}" # convert to lowercase
+        if is_valid_arg "mongo mysql edxapp fullstack" $EDX_ROLE ; then
+          shift # past argument
+        else
+          echo "Invalid role specified\n"
+          display_usage
+        fi
+        ;;
+      -e|--environment)
+        DEPLOYMENT_ENV="${2,,}" # convert to lowercase
+        if is_valid_arg "dev bvt int prod" $DEPLOYMENT_ENV ; then
+          shift # past argument
+        else
+          echo "Invalid environment specified\n"
+          display_usage
+        fi
+        ;;
+      -a|--access_token)
+        ACCESS_TOKEN="$2"
+        if [[ ${#ACCESS_TOKEN} -eq 40 ]] ; then
+          shift # past argument
+        else
+          echo "Invalid access token specified\n"
+          display_usage
+        fi
+        ;;
+      *)
+        # Unknown option encountered
+        display_usage
+        ;;
+    esac
+
+    shift # past argument or value
+  done
+}
+
 setup() {
   if [[ ! -d $OXA_CONFIG_PATH ]]; then
     wget https://raw.githubusercontent.com/edx/configuration/master/util/install/ansible-bootstrap.sh -O - | bash
 
     # must match $OXA_TOOLS_PATH/config/edx-versions.yml for now
-    CONFIGURATION_REPO=https://github.com/Microsoft/edx-configuration.git
-    CONFIGURATION_VERSION="lex/scalable-dogwood"
+    local CONFIGURATION_REPO=https://github.com/Microsoft/edx-configuration.git
+    local CONFIGURATION_VERSION="lex/scalable-dogwood"
 
     git clone $CONFIGURATION_REPO $OXA_CONFIG_PATH
     cd $OXA_CONFIG_PATH
     git checkout $CONFIGURATION_VERSION
 
     pip install -r requirements.txt
   fi
-  
+
+  # Remove the oxa-tools-config directory if it exists to avoid the error
+  # "fatal: destination path 'oxa-tools-config' already exists and is not an empty directory."
+  if [[ -d $OXA_TOOLS_PATH/oxa-tools-config ]]; then
+    sudo rm -R $OXA_TOOLS_PATH/oxa-tools-config
+  fi
+
+  # Fetch the latest secrets from the private repo via a personal access token
+  cd $OXA_TOOLS_PATH
+  sudo git clone https://$ACCESS_TOKEN@github.com/microsoft/oxa-tools-config.git
```

Comment:
Instead of removing and re-cloning each time, would just clone on first time. Would also do checkout of master each time. Soon, once we have prod/int environments and release branch, we'll need to account for that.